conformance/tests/httproute-invalid-reference-policy.go

```diff
@@ -0,0 +1,113 @@
+		routeNN := types.NamespacedName{Name: "invalid-reference-policy", Namespace: "gateway-conformance-infra"}
+		gwNN := types.NamespacedName{Name: "same-namespace", Namespace: "gateway-conformance-infra"}
+
+		ns := v1alpha2.Namespace(gwNN.Namespace)
+		gwKind := v1alpha2.Kind("Gateway")
+
+		// TODO(mikemorris): Add check for Accepted condition once
+		// https://github.com/kubernetes-sigs/gateway-api/issues/1112
+		// has been resolved
+		t.Run("Route status should have a route parent status with a ResolvedRefs condition with status False and reason RefNotPermitted", func(t *testing.T) {
+			parents := []v1alpha2.RouteParentStatus{{
+				ParentRef: v1alpha2.ParentReference{
+					Group:     (*v1alpha2.Group)(&v1alpha2.GroupVersion.Group),
+					Kind:      &gwKind,
+					Name:      v1alpha2.ObjectName(gwNN.Name),
+					Namespace: &ns,
+				},
+				ControllerName: v1alpha2.GatewayController(s.ControllerName),
+				Conditions: []metav1.Condition{{
+					Type:   string(v1alpha2.RouteConditionResolvedRefs),
+					Status: metav1.ConditionFalse,
+					Reason: string(v1alpha2.RouteReasonRefNotPermitted),
+				}},
+			}}
+
+			kubernetes.HTTPRouteMustHaveParents(t, s.Client, routeNN, parents, false, 60)
+		})
+
+		// TODO(mikemorris): Un-skip check for Listener ResolvedRefs
+		// RefNotPermitted condition and/or add check for attached
+		// routes and any expected Listener conditions once
+		// https://github.com/kubernetes-sigs/gateway-api/issues/1112
+		// has been resolved
+		t.Skip("Gateway listener should have a ResolvedRefs condition with status False and reason RefNotPermitted", func(t *testing.T) {
+			listeners := []v1alpha2.ListenerStatus{{
+				Name: v1alpha2.SectionName("http"),
+				SupportedKinds: []v1alpha2.RouteGroupKind{{
+					Group: (*v1alpha2.Group)(&v1alpha2.GroupVersion.Group),
+					Kind:  v1alpha2.Kind("HTTPRoute"),
+				}},
+				Conditions: []metav1.Condition{{
+					Type:   string(v1alpha2.RouteConditionResolvedRefs),
+					Status: metav1.ConditionFalse,
+					Reason: string(v1alpha2.RouteReasonRefNotPermitted),
+				}},
+			}}
+
+			kubernetes.GatewayStatusMustHaveListeners(t, s.Client, gwNN, listeners, 60)
+		})
+
+		gwAddr := kubernetes.GatewayAndHTTPRoutesMustBeReady(t, s.Client, s.ControllerName, gwNN, routeNN)
+
+		// TODO(mikemorris): Add check for HTTP requests successfully reaching
+		// app-backend-v1 at path "/" if it is determined that a Route with at
+		// at least one allowed BackendRef should be accepted by a Gateway
+		// and partially configured.
+
+		t.Run("Simple HTTP request should not reach app-backend-v2", func(t *testing.T) {
+			http.MakeRequestAndExpectEventuallyConsistentResponse(t, s.RoundTripper, gwAddr, http.ExpectedResponse{
+				Request: http.ExpectedRequest{
+					Method: "GET",
+					Path:   "/v2",
+				},
+				StatusCode: 503,
+				// TODO: should these fields be populated when the BackendRef is invalid?
+				Backend:   "app-backend-v2",
+				Namespace: "gateway-conformance-app-backend",
```

Comment:
I missed this earlier, but I think these lines should be removed since they won't be checked in any way (https://github.com/kubernetes-sigs/gateway-api/blob/master/conformance/utils/http/http.go#L137).

Reply:
```suggestion
```